### PR TITLE
remove okteto-compose specific extentions to demo

### PR DESCRIPTION
**Pull request by gabriel-okteto**

### CODE REVIEW

**Review by pchico83 — APPROVED** [review]

File: api/server.js [hunk 1]
```diff
@@ -21,6 +22,9 @@ function startWithRetry() {
     const db = client.db(process.env.MONGODB_DATABASE);
 
     app.listen(8080, () => {
+
+      app.use(cors());
```
Comment (pchico83) [inline api/server.js:26]: no need to manage cors, it's one of the advantages of using okteto

File: api/server.js [hunk 2]
```diff
@@ -40,7 +44,7 @@ function startWithRetry() {
 
       app.get("/api/watching", (req, res, next) => {
         console.log(`GET /api/watching`)
-        db.collection('movies').find().toArray( (err, results) =>{
+        db.collection('watching').find().toArray( (err, results) =>{
```
Comment (pchico83) [inline api/server.js:47]: undo this one, it's the bug we want to fix as part of the getting started guide

File: docker-compose.yml [hunk 3]
```diff
@@ -1,32 +1,26 @@
 name: movies
```
Comment (pchico83) [inline docker-compose.yml:1]: remove `name`, it's not a docker-compose thing

File: docker-compose.yml [hunk 4]
```diff
@@ -1,32 +1,26 @@
...
-    port: 8080
-
 services:
 
   frontend:
+    scale: 8
     build: frontend
     ports:
-      - 80
+      - 80:80
```
Comment (pchico83) [inline docker-compose.yml:9]: this should be `80`. that way, it's not exposed publicly. the only public endpoint should be for a new `nginx` service in the compose file

File: docker-compose.yml [hunk 5]
```diff
@@ -1,32 +1,26 @@
...
-    path: /
-    port: 80
-  - service: api
-    path: /api
-    port: 8080
-
 services:
 
   frontend:
+    scale: 8
```
Comment (pchico83) [inline docker-compose.yml:6]: undo before final merge

File: docker-compose.yml [hunk 6]
```diff
@@ -1,32 +1,26 @@
...
     build: frontend
     ports:
-      - 80
+      - 80:80
     volumes:
       - frontend:/src
     command: yarn start
 
   api:
+    scale: 4
```
Comment (pchico83) [inline docker-compose.yml:15]: undo before final merge

File: docker-compose.yml [hunk 7]
```diff
@@ -1,32 +1,26 @@
...
     build: api
     environment:
       - MONGODB_USERNAME=okteto
       - MONGODB_PASSWORD=mongodb123
       - MONGODB_DATABASE=okteto
       - MONGODB_HOST=mongodb
     ports:
-      - 8080
+      - 8080:8080
```
Comment (pchico83) [inline docker-compose.yml:23]: undo this change

File: frontend/src/App.jsx [hunk 8]
```diff
@@ -31,7 +31,7 @@ class App extends Component {
   }
 
   componentDidMount() {
-    fetch('/api/movies')
+    fetch('https://api-gabriel-okteto.cloud.okteto.net/api/movies')
```
Comment (pchico83) [inline frontend/src/App.jsx:34]: undo, this should be '/api/movies'

File: frontend/src/App.jsx [hunk 9]
```diff
@@ -42,7 +42,7 @@ class App extends Component {
         });
       });
 
-    fetch('/api/watching')
+    fetch('https://api-gabriel-okteto.cloud.okteto.net/api/watching')
```
Comment (pchico83) [inline frontend/src/App.jsx:45]: same

File: frontend/webpack.config.js [hunk 10]
```diff
@@ -63,7 +63,7 @@ module.exports = {
       poll: true
     },
     proxy: {
-      '/api': 'http://movies-api:8080'
+      '/api': 'https://api-gabriel-okteto.cloud.okteto.net'
```
Comment (pchico83) [inline frontend/webpack.config.js:66]: do we need a proxy at all?

**Review by pchico83 — CHANGES_REQUESTED** [review]
@gabriel-okteto the approach is wrong. the idea is to have a new service in the compose based on nginx. this container redirects to the frontend or the API container based on the path. but the code for frontend and API should be unaffected by this change

File: docker-compose.yml (hunk 6 above)
Comment (gabriel-okteto) [reply docker-compose.yml:20]: completely and remove the line or set to 1 or ...

File: docker-compose.yml [hunk 11]
```diff
@@ -1,16 +1,14 @@
...
-
 services:
 
+  movies:
+    build: reverse-proxy
+    ports:
+      - 80:80
+
   frontend:
+    scale: 1
```
Comment (pchico83) [inline docker-compose.yml:11]: I would remove this one, it's redundant. and deprecated in the compose spec

Comment (gabriel-okteto) [reply docker-compose.yml:11]: awee.. its 100% compatible and shows one of our strengths 

also, yes I will remove it

File: docker-compose.yml [hunk 12]
```diff
@@ -0,0 +1,65 @@
+
+x-common-variables: &common-variables
```
Comment (pchico83) [inline docker-compose.yml]: let's move this to a `.env` file. It what's most people do for envvars

File: docker-compose.yml [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+services:
+
+  movies:
+    build: reverse-proxy
+    ports:
+      - 80:80
+    depends_on:
+      - api
+      - frontend
+    restart: always
```
Comment (pchico83) [inline docker-compose.yml:11]: remove this one, it's the default value

**Review by pchico83 — APPROVED** [review]

File: docker-compose.yml (hunk 12 above)
Comment (gabriel-okteto) [reply docker-compose.yml]: i reverted to the previous longhand
